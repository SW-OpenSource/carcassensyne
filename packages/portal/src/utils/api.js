// import axios from 'axios'
import { Auth } from 'utils'

const { DEV_API_URL, PROD_API_URL, NODE_ENV } = process.env
const API_URL = NODE_ENV === 'development' ? DEV_API_URL : PROD_API_URL

const api = query =>
  fetch(API_URL, {
    method: 'POST',
    body: JSON.stringify(query),
    headers: {
      Authorization: `bearer ${Auth.getToken()}`,
      'Content-Type': 'application/json',
    },
  })

export { api }
