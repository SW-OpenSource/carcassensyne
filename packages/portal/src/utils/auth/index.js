import decode from 'jwt-decode'
import * as CONSTS from './consts'

const Auth = {
  // IS
  isToken: () => typeof localStorage.getItem(CONSTS.TOKEN) === 'string',
  // GET
  getToken: () => (Auth.isToken() ? localStorage.getItem(CONSTS.TOKEN) : undefined),
  getIdFromToken: () => decode(Auth.getToken())._id,
  // SET
  setToken: token => {
    localStorage.setItem(CONSTS.TOKEN, token)
  },
  // REMOVE
  removeToken: () => localStorage.removeItem(CONSTS.TOKEN),
  handleLogin: (response, history) => {
    const res = { ...response }
    if (res.data) {
      // console.group('%cAuth handleLogin', 'color: green')
      // console.log({ response })
      // console.groupEnd()
      if (typeof res.data.login === 'string') {
        Auth.setToken(res.data.login)
        history.replace('/')
      }
    } else if (res.error) {
      // const token = response.data.login
      // console.group('%cAuth handleLogin', 'color: tomato')
      // console.log({ response })
      // console.groupEnd()
    }
  },
  handleRegistration: (response, history) => {
    const res = { ...response }
    if (res.data) {
      // console.group('%cAuth handleLogin', 'color: green')
      // console.log({ response })
      // console.groupEnd()
      if (typeof res.data.register === 'string') {
        Auth.setToken(res.data.register)
        history.replace('/')
      }
    } else if (res.error) {
      // const token = response.data.login
      // console.group('%cAuth handleLogin', 'color: tomato')
      // console.log({ response })
      // console.groupEnd()
    }
  },
  validateToken: () => {
    if (Auth.isToken()) {
      const { exp } = decode(Auth.getToken())
      if (exp > new Date().getTime() / 1000) {
        return true
      }
    }
    return false
  },
  handleLogout: history => {
    Auth.removeToken()
    history.push('/')
  },
}

export { Auth }
