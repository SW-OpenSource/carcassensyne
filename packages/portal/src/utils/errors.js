const DUPLICATE_EMAIL = 'email_1 dup key'
const DUPLICATE_UERNAME = 'username_1 dup key'

const registrationErrors = string => {
  if (string.includes(DUPLICATE_EMAIL)) {
    return 'Email address already in use!'
  }
  if (string.includes(DUPLICATE_UERNAME)) {
    return 'Username already in use!'
  }
}

const getErrorField = string => {
  if (string.includes(DUPLICATE_EMAIL)) {
    console.log(' error email ')
    return 'email'
  }
  if (string.includes(DUPLICATE_UERNAME)) {
    console.log(' error username ')
    return 'username'
  }
}

export { registrationErrors, getErrorField }
