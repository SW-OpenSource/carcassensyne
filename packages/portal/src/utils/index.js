export { Auth } from './auth'
export { api } from './api'
export { Age } from './age'
