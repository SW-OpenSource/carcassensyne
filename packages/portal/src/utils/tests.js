import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import { theme } from '../resources/styles/theme'

const renderWithTheme = component =>
  renderer.create(<ThemeProvider theme={theme}>{component}</ThemeProvider>)

export { renderWithTheme }
