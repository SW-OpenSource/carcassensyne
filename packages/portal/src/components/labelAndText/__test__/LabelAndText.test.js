import 'jest-styled-components'
import React from 'react'
import { mount } from 'enzyme'
import { LabelAndText } from '../index'
import { renderWithTheme } from 'utils/tests'

describe('LabelAndText', () => {
  test('Should render a Text element', () => {
    const component = renderWithTheme(
      <LabelAndText
        label="Hello World!"
        text={
          <span role="img" aria-label="Hello World!">
            🙌
          </span>
        }
      />,
    ).toJSON()
    expect(component).toMatchSnapshot()
  })

  test('Should render correct label', () => {
    const EXPECTED_RESULT = 'Hello World!'
    const component = mount(
      <LabelAndText
        label="Hello World!"
        text={
          <span role="img" aria-label="Hello World!">
            🙌
          </span>
        }
      />,
    )
    expect(component.props().label).toStrictEqual(EXPECTED_RESULT)
  })

  test('Should render correct text', () => {
    const EXPECTED_RESULT = (
      <span role="img" aria-label="Hello World!">
        🤘🏻
      </span>
    )
    const component = mount(
      <LabelAndText
        label="Hello World!"
        text={
          <span role="img" aria-label="Hello World!">
            🤘🏻
          </span>
        }
      />,
    )
    expect(component.props().text).toStrictEqual(EXPECTED_RESULT)
  })
})
