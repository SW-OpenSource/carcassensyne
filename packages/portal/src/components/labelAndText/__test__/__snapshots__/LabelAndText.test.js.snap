// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelAndText Should render a Text element 1`] = `
.c1 {
  color: #333;
  font-size: 1em;
  font-weight: 600;
}

.c2 {
  color: #333;
  font-size: 1em;
  font-weight: 400;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  text-align: left;
}

.c0 > p:first-of-type {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
  className="c0"
>
  <p
    className="c1"
    color="#333"
  >
    Hello World!
  </p>
  <p
    className="c2"
    color="#333"
  >
    <span
      aria-label="Hello World!"
      role="img"
    >
      ðŸ™Œ
    </span>
  </p>
</div>
`;
